### PR TITLE
Enhancement to minigraph parsing for chassis/multi-asic use case

### PR DESCRIPTION
**Pull request by abdosi**
What I did:

Following changes are done:     

1. Added Support where if asic configuration is not present in minigraph sonic-cfggen do not error out but instead process it gracefully. 
`Use Case: In Supervisor we have number of asic are define as max possible but in minigraph configuration of only valid/available asics only are present. Without this change `load_minigraph` fails.`

Microsoft ADO: https://msazure.visualstudio.com/One/_workitems/edit/17956325

How I Verify:
UT has been added.
Manual Verification

### CODE REVIEW

**Comment by abdosi** [comment]
@arlakshm / @judyjoseph  can you please help review this.

File: src/sonic-config-engine/minigraph.py [hunk 1]
```diff
@@ -1804,15 +1833,15 @@ def parse_xml(filename, platform=None, port_config_file=None, asic_name=None, hw
     results['PORTCHANNEL_INTERFACE'] = pc_intfs
 
     # for storage backend subinterface info present in minigraph takes precedence over ResourceType
-    if current_device['type'] in backend_device_types and bool(vlan_sub_intfs):
+    if current_device and current_device['type'] in backend_device_types and bool(vlan_sub_intfs):
```
Comment (judyjoseph) [inline src/sonic-config-engine/minigraph.py:1836]: To add check for current_device in line :1567

    # for this hostname, if sub_role is defined, add sub_role in 
    # device_metadata
    if sub_role is not None:
        current_device['sub_role'] = sub_role
        results['DEVICE_METADATA']['localhost']['sub_role'] =  sub_role
        results['DEVICE_METADATA']['localhost']['asic_name'] =  asic_name

Comment (abdosi) [reply src/sonic-config-engine/minigraph.py:1836]: @judyjoseph sub_role will be none if current_Device is not there so we should be fine.

**Review by judyjoseph — APPROVED** [review]

**Comment by abdosi** [comment]
@rlhui Can you please help merge this.

**Comment by lguohan** [comment]
can we separate this into three PRs, one for each issue? code are mixed together, i am not able to differentiate the 1 and 3 issue addressed.

**Comment by abdosi** [comment]
> can we separate this into three PRs, one for each issue? code are mixed together, i am not able to differentiate the 1 and 3 issue addressed.

will do.

**Comment by abdosi** [comment]
@judyjoseph can you please re-approve. i had separated the changes in multiple PR's.

**Review by judyjoseph — APPROVED** [review]
LGTM

**Comment by abdosi** [comment]
@lguohan can you please help merge of this

**Comment by liushilongbuaa** [comment]
@abdosi , code conflict with internal branch.
Please help resolve.

**Pull request merged**